crypto/secp256k1/secp256.go

```diff
@@ -49,6 +49,41 @@ var (
 	ErrRecoverFailed       = errors.New("recovery failed")
 )
 
+func VerifySeckeyValidity(seckey []byte) error {
+	if len(seckey) != 32 {
+		return errors.New("priv key is not 32 bytes")
+	}
+	var seckey_ptr *C.uchar = (*C.uchar)(unsafe.Pointer(&seckey[0]))
+	ret := C.secp256k1_ec_seckey_verify(context, seckey_ptr)
+	if int(ret) != 1 {
+		return errors.New("invalid seckey")
+	}
+	return nil
+}
+
+func GeneratePubKey(seckey []byte) ([]byte, error) {
+	if err := VerifySeckeyValidity(seckey); err != nil {
+		return nil, err
+	}
+
+	var pubkey []byte = make([]byte, 64)
+	var pubkey_ptr *C.secp256k1_pubkey = (*C.secp256k1_pubkey)(unsafe.Pointer(&pubkey[0]))
+
+	var seckey_ptr *C.uchar = (*C.uchar)(unsafe.Pointer(&seckey[0]))
+
+	ret := C.secp256k1_ec_pubkey_create(
```

Comment:
The result is different because `ec_pubkey_create` deals with libsecp256k1 internal public key representation, which uses native integers for the EC point values. To make this work, we'd need to call `ec_pubkey_serialize` on the result.